packages/cdktf/lib/terraform-backend.ts

```diff
@@ -1,17 +1,34 @@
 import { Construct } from "constructs";
+import { TerraformRemoteState } from "./terraform-remote-state";
 import { TerraformElement } from "./terraform-element";
 import { deepMerge } from "./util";
 
+const BACKEND_SYMBOL = Symbol.for("cdktf/TerraformBackend");
+
 export abstract class TerraformBackend extends TerraformElement {
   constructor(scope: Construct, id: string, protected readonly name: string) {
     super(scope, id);
+    Object.defineProperty(this, BACKEND_SYMBOL, { value: true });
+  }
+
+  public static isBackend(x: any): x is TerraformBackend {
+    return x !== null && typeof x === "object" && BACKEND_SYMBOL in x;
   }
 
   // jsii can't handle abstract classes?
   protected synthesizeAttributes(): { [name: string]: any } {
     return {};
   }
 
+  /**
+   * Creates a TerraformRemoteState resource that accesses this backend.
+   */
+  public abstract getRemoteStateDataSource(
+    _scope: Construct,
+    _name: string,
+    _fromStack: string
```

Comment:
Are underscores needed for "unused" parameters of abstract methods?

Reply:
Huh, apparently not. I think I had it non-abstract before and didn't change the args